src/include/server/mir/scene/idle_hub.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2021 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored By: William Wold <william.wold@canonical.com>
+ */
+
+#ifndef MIR_SCENE_IDLE_HUB_H_
+#define MIR_SCENE_IDLE_HUB_H_
+
+#include "mir/time/types.h"
+
+#include <memory>
+#include <optional>
+#include <string>
+
+namespace mir
+{
+class Executor;
+namespace scene
+{
+/// Gets notifications about changes in the idle state
```

Comment:
This comment is wrong.
```suggestion
/// Gets notifications about the idle state when registered and subsequent state changes until unregistered
```